main.tf

```diff
@@ -8,9 +8,25 @@ terraform {
     required_version = ">= 1.2.0"
 }
 
+resource "random_password" "password" {
```

Comment:
also if your project is a small, you should name your resources as "this", because with this approach you may clearly see relations between different resources (clear binding)

also, in the next tasks, I would recommend to create different files for different resources, for example for ASG - asg.tf, ALB -alb.tf, EKS - eks.tf etc...